### PR TITLE
modify sequencingLoop in sequencer

### PR DESCRIPTION
**Pull request by kwhuo68**
Core changes:
-Modifies sequencingLoop to use a timer
-Each batch gets added to a batchTxs, and then every timer interval those batches get processed

### CODE REVIEW

File: clients/geth/specular/rollup/services/sequencer/sequencer.go [hunk 1]
```diff
@@ -142,31 +147,47 @@ func (s *Sequencer) sequencingLoop(genesisRoot common.Hash) {
...
+				queuedAssertion.EndBlock = currBatchTxs.LastBlockNumber()
+				// If no assertion is pending, commit it
+				if pendingAssertion == nil {
+					commitAssertion()
+				}
 			}
+
+			// Clear batches
+			batchTxs = nil
+			timer.Reset(timeInterval)
```
Comment (LEAFERx) [inline clients/geth/specular/rollup/services/sequencer/sequencer.go:186]: Why do you need to reset a ticker?

File: clients/geth/specular/rollup/services/sequencer/sequencer.go [hunk 2]
```diff
@@ -142,31 +147,47 @@ func (s *Sequencer) sequencingLoop(genesisRoot common.Hash) {
...
-			// New batch from Batcher
-			log.Debug("Batch", "length", len(batch.Txs))
-			// Sequence batch to SequencerInbox
-			contexts, txLengths, txs, err := batch.SerializeToArgs()
-			if err != nil {
-				log.Error("Can not serialize batch", "error", err)
-				continue
+		case <-timer.C:
+			if len(batchTxs) == 0 {
+				timer.Reset(timeInterval)
```
Comment (LEAFERx) [inline clients/geth/specular/rollup/services/sequencer/sequencer.go:157]: Why you need to reset a ticker?

**Review by LEAFERx — CHANGES_REQUESTED** [review]

**Review by LEAFERx — APPROVED** [review]

File: clients/geth/specular/rollup/services/sequencer/sequencer.go [hunk 3]
```diff
@@ -96,11 +97,26 @@ func (s *Sequencer) batchingLoop() {
 	}
 }
 
-// This goroutine sequences batches to L1 SequencerInbox and creates DAs
-// TODO: create assertions by timer
+// Combines batches
+func combineBatches(slice []*rollupTypes.TxBatch) *rollupTypes.TxBatch {
+	var blocks []*types.Block
+
+	for i := 0; i <= len(slice)-1; i++ {
```
Comment (ujvl) [inline clients/geth/specular/rollup/services/sequencer/sequencer.go:104]: does gofmt enforce no spacing around arithmetic like this, that's annoying

File: clients/geth/specular/rollup/services/sequencer/sequencer.go [hunk 4]
```diff
@@ -96,11 +97,26 @@ func (s *Sequencer) batchingLoop() {
...
+	}
+	combinedBatch := rollupTypes.NewTxBatch(blocks, 0)
+	return combinedBatch
+}
+
 func (s *Sequencer) sequencingLoop(genesisRoot common.Hash) {
 	defer s.Wg.Done()
 
+	// Timer
+	var timeInterval = 10 * time.Second
```
Comment (ujvl) [inline clients/geth/specular/rollup/services/sequencer/sequencer.go:116]: nit: define const `TIME_INTERVAL` globally

File: clients/geth/specular/rollup/services/sequencer/sequencer.go [hunk 5]
```diff
@@ -159,14 +179,20 @@ func (s *Sequencer) sequencingLoop(genesisRoot common.Hash) {
...
+			queuedAssertion.VmHash = combinedBatch.LastBlockRoot()
+			queuedAssertion.CumulativeGasUsed.Add(queuedAssertion.CumulativeGasUsed, combinedBatch.GasUsed)
+			queuedAssertion.InboxSize.Add(queuedAssertion.InboxSize, combinedBatch.InboxSize())
+			queuedAssertion.EndBlock = combinedBatch.LastBlockNumber()
 			// If no assertion is pending, commit it
 			if pendingAssertion == nil {
 				commitAssertion()
 			}
+			batchTxs = nil
+			combinedBatch = nil
```
Comment (ujvl) [inline clients/geth/specular/rollup/services/sequencer/sequencer.go:191]: probably makes more sense to just declare `var combinedBatch *rollupTypes.TxBatch` in the case (then it'll also be GC'd automatically w/o assigning nil)

**Review by ujvl — APPROVED** [review]
couple minor comments but otherwise lgtm!

**Pull request merged**